src/Core/Domain/Carrier/Command/AddCarrierCommand.php

```diff
@@ -0,0 +1,360 @@
+     */
+    public function getShippingRanges(): array
+    {
+        return $this->shippingRanges;
+    }
+
+    /**
+     * @return int
+     */
+    public function getTaxRulesGroupId(): int
+    {
+        return $this->taxRulesGroupId;
+    }
+
+    /**
+     * @return OutOfRangeBehavior
+     */
+    public function getOutOfRangeBehavior(): OutOfRangeBehavior
+    {
+        return $this->outOfRangeBehavior;
+    }
+
+    /**
+     * @return int
+     */
+    public function getMaxPackageWidth(): int
+    {
+        return $this->maxPackageWidth;
+    }
+
+    /**
+     * @return int
+     */
+    public function getMaxPackageHeight(): int
+    {
+        return $this->maxPackageHeight;
+    }
+
+    /**
+     * @return int
+     */
+    public function getMaxPackageDepth(): int
+    {
+        return $this->maxPackageDepth;
+    }
+
+    /**
+     * @return float
+     */
+    public function getMaxPackageWeight(): float
+    {
+        return $this->maxPackageWeight;
+    }
+
+    /**
+     * @return int[]
+     */
+    public function getAssociatedGroupIds(): array
+    {
+        return $this->associatedGroupIds;
+    }
+
+    /**
+     * @return int[]
+     */
+    public function getAssociatedShopIds(): array
+    {
+        return $this->associatedShopIds;
+    }
+
+    /**
+     * @param array $shippingRanges
+     *
+     * @throws CarrierConstraintException
+     */
+    private function setShippingRanges(array $shippingRanges)
+    {
+        foreach ($shippingRanges as $range) {
+            $this->shippingRanges[] = ShippingRange::buildFromArray($range);
+        }
+    }
+
+    /**
+     * @param array $localizedCarrierNames
+     *
+     * @throws CarrierConstraintException
+     */
+    private function setLocalizedCarrierNames(array $localizedCarrierNames)
+    {
+        foreach ($localizedCarrierNames as $langId => $name) {
+            $this->localizedCarrierNames[(new LanguageId($langId))->getValue()] = new CarrierName($name);
+        }
+    }
+
+    /**
+     * @param array $localizedShippingDelays
+     *
+     * @throws CarrierConstraintException
+     */
+    private function setLocalizedShippingDelays(array $localizedShippingDelays)
+    {
+        foreach ($localizedShippingDelays as $langId => $delay) {
+            $this->localizedShippingDelays[(new LanguageId($langId))->getValue()] = new ShippingDelay($delay);
+        }
+    }
+
+    /**
+     * @param int $width
+     * @param int $height
+     * @param int $depth
+     * @param float $weight
+     *
+     * @throws CarrierConstraintException
+     */
+    private function setMeasures(int $width, int $height, int $depth, float $weight)
+    {
+        foreach ([$width, $height, $depth, $weight] as $measure) {
+            $this->assertMeasureIsNonNegative($measure);
+        }
+
+        $this->maxPackageWidth = $width;
+        $this->maxPackageHeight = $height;
+        $this->maxPackageDepth = $depth;
+        $this->maxPackageWeight = $weight;
```

Comment:
Looks like it could be VO?

Reply:
I had created 2 VO's for this case, the PackgageSizeMeasure (for int value) and PackageWeightMeasure (for float values), but these seemed to be overhead when i started using strict types, as it only asserts for non-negative value inside.